src/StringToVectorConverter.cpp

```diff
@@ -0,0 +1,45 @@
+#include "StringToVectorConverter.hpp"
+
+int lambda(char c){
+    if(c == '-'){
+        return 0;
+    }else if(std::isdigit(c) ){
+        return c - 48;
+    }else if(c == 'X') {
+        return 10;
+    }
+    return 0;
+}
+
+std::string ExtractPointsFromString(std::string playerFrames){
+    if(playerFrames.find(':') != std::string::npos)
+        playerFrames.erase(playerFrames.begin(), std::find(std::begin(playerFrames), std::end(playerFrames), (':'))+1);
+
+    playerFrames.erase(std::remove(playerFrames.begin(), playerFrames.end(), '|'),playerFrames.end());
+
+    for(int i = 0; i < playerFrames.length(); i ++)
+        if(playerFrames[i] == '/')
+            playerFrames[i] = *std::to_string( 10 - (static_cast<int>(playerFrames[i-1]) - 48) ).c_str();
```

Comment:
poor readability. Extract to a separate function with a proper name or simplify it.